### DISCUSSION TITLE
Template for grouping PRs of minor/patches

### DISCUSSION THREAD

**Discussion by ryanmerolle** [renovatebot/renovate#13226] [Request Help]
Long time admirer here getting off the ground with renovate for some public repos on gitlab.  As a part time maintainer of a few projects myself, I really appreciate the community and work done with this project.

I have been combing the docs, and I am sort of confused on how to tackle renovate.json config for:

- Grouping into 1 PR, all minor releases & patch releases based on semantic versioning. (All major releases being their own PR for testing)
- Grouping by package manager (not something I will likely due, given I hope to test and implement the above.  Something I want to test workflow wise regardless)

If I figure it out from the docs, I'll post back here.  As always, any guidance or templates are greatly appreciated!

**Comment by gomesp** [comment]
Hi @ryanmerolle I am using the example listed [in this preset](https://docs.renovatebot.com/presets-group/#groupallnonmajor) for grouping minor&patch releases. Grouping by package manager would be really cool indeed for monorepos.

**Reply by rarkins** [reply]
A simple way of thinking about grouping is: if the branch is the same, they will be grouped (whether intentional or not).

One of the options for this is to utilize `additionalBranchPrefix`. e.g. if you configure `"additionalBranchPrefix": "{{manager}}-"` then it will mean you have branches like `renovate/npm-lodash-4.x` or `renovate/npm-all-non-major`.

**Reply by gomesp** [reply]
Hi @rarkins, thanks for your answer. It doesn't seem that ```{{manager}}``` is an  exposed [template field](https://docs.renovatebot.com/templates/#additional-handlebars-helpers). At least adding the following to my config didn't change anything:

```
  packageRules: [
    {
      matchPackagePatterns: ["*"],
      matchUpdateTypes: ["minor", "patch"],
      groupName: "all non-major dependencies",
      groupSlug: "all-minor-patch",
      additionalBranchPrefix: "{{manager}}-",
    },
  ],

```

Thanks

**Reply by gomesp** [reply]
Looking at the documentation I imagine that using the following setting would achieve what you want: ```additionalBranchPrefix: "{{packageFile}}-",```

**Reply by rarkins** [reply]
Sorry, you're right - looks like we haven't exposed `manager` for templating according to this: https://docs.renovatebot.com/templates/

``{{datasource}}-` may be more concise for you
